apis/field_error.go

```diff
@@ -332,6 +332,16 @@ func ErrInvalidValue(value interface{}, fieldPath string) *FieldError {
 	}
 }
 
+// ErrInvalidValueWithDetails constructs a FieldError for a field that has received an
+// invalid value including Details.
+func ErrInvalidValueWithDetails(value interface{}, fieldPath string, details string) *FieldError {
```

Comment:
I think it would be simpler to modify `ErrInvalidValue` to take an optional variadic input for the details strings - ie. see `ErrInvalidKeyName`

https://github.com/knative/pkg/blob/dabb207d45983cfd03bd10985639b42c0b3db404/apis/field_error.go#L374-L380